quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/PropertyMappingInterceptor.java

```diff
@@ -38,6 +46,57 @@
  */
 public class PropertyMappingInterceptor implements ConfigSourceInterceptor {
 
+    <T> Iterator<T> filterRuntime(Iterator<T> iter, Function<T, String> nameFunc) {
+        if (!isRebuild() && !Environment.isRebuildCheck()) {
+            return iter;
+        }
+        return new Iterator<T>() {
+            T next;
+            boolean isNull;
+
+            @Override
+            public boolean hasNext() {
+                while (next == null && !isNull) {
+                    if (!iter.hasNext()) {
+                        return false;
+                    }
+                    T value = iter.next();
+                    if (!isRuntime(nameFunc.apply(value))) {
+                        next = value;
+                        isNull = next == null;
+                    }
+                }
+                return true;
+            }
```

Comment:
@shawkins What about this? We'd remove the isNull var and the functionality would be the same, right? 
When the runtime option is present, it just ignores it. If the build option, or null, is present, it returns true.
WDYT?

```suggestion
            @Override
            public boolean hasNext() {
                while (iter.hasNext()) {
                    T value = iter.next();
                    if (isRuntime(nameFunc.apply(value))) continue;
                    next = value;
                    return true;
                }
                return false;
            }
```